File: 2/day2.scala
import scala.io.Source

/**
  * Created by mzimmerman on 1/16/17.
  */


def calc(input: Iterable[String]) = {
  var totalpaper = 0
  var totalribbon = 0
  for (line <- input) {
    val dim = line.split("x").take(3).map(_.toInt).sorted
    val area = Seq(dim(0) * dim(1), dim(1) * dim(2), dim(2) * dim(0)).sorted
    val paper = 2*area(0) + 2*area(1) + 2*area(2) + area(0)
    totalpaper += paper

    val vol = dim(0) * dim(1) * dim(2)
    val ribbon = 2*dim(0) + 2*dim(1) + vol
    totalribbon += ribbon
  }

  println(s"total paper = $totalpaper")
  println(s"total ribbon = $totalribbon")
}

val input = Source.fromFile("input2.txt").getLines().toList
calc(input)

File: 2/day2.py
#!/usr/bin/python

input="input2.txt"

fh=open(input,"r")
total=0
for line in fh:
    line=line.rstrip()
    #print line
    (x, y, z) = [int(i) for i in line.split("x")]
    area = sorted([x*y, y*z, x*z])
    totalarea = 2*area[0] + 2*area[1] + 2*area[2] + minarea
    total += totalarea
    print "dim: %2d %2d %2d areas: %4d %4d %4d minarea: %4d totalarea: %5d TOTAL: %7d" % (x, y, z, area[0], area[1], area[2], minarea, totalarea, total)
